elephant/sta.py

```diff
@@ -0,0 +1,146 @@
+    else:
+        raise TypeError(
+            "spiketrains must be a SpikeTrain, a numpy ndarray, or a list of one of those, not %s." % type(spiketrains))
+
+    # multiplying spiketrain in case only a single spiketrain is given
+    if len(spiketrains) == 1 and num_signals != 1:
+        template = spiketrains[0]
+        spiketrains = []
+        for i in range(num_signals):
+            spiketrains.append(template)
+
+    # checking for matching numbers of signals and spiketrains
+    if num_signals != len(spiketrains):
+        raise ValueError(
+            "The number of signals and spiketrains has to be the same.")
+
+    # checking the times of signal and spiketrains
+    for i in range(num_signals):
+        if spiketrains[i].t_start < signal.t_start:
+            raise ValueError(
+                "The spiketrain indexed by %i starts earlier than the analog signal." % i)
+        if spiketrains[i].t_stop > signal.t_stop:
+            raise ValueError(
+                "The spiketrain indexed by %i stops later than the analog signal." % i)
+
+    # *** Main algorithm: ***
+
+    # window_bins: number of bins of the chosen averaging interval
+    window_bins = int(np.round(
+        ((window_stoptime - window_starttime) * signal.sampling_rate).simplified))
+    # result_sta: array containing finally the spike-triggered averaged signal
+    result_sta = AnalogSignalArray(np.zeros(
+        (window_bins, num_signals)), sampling_rate=signal.sampling_rate, units=signal.units)
+    # setting of correct times of the spike-triggered average relative to the spike
+    result_sta.t_start = window_starttime
+    used_spikes = np.zeros(num_signals, dtype=int)
+    unused_spikes = np.zeros(num_signals, dtype=int)
+    total_used_spikes = 0
+
+    for i in range(num_signals):
+        # summing over all respective signal intervals around spiketimes
+        for spiketime in spiketrains[i]:
+            # checks for sufficient signal data around spiketime
+            if spiketime + window_starttime >= signal.t_start and spiketime + window_stoptime <= signal.t_stop:
+                # calculating the startbin in the analog signal of the averaging window for spike
+                startbin = int(np.round(
+                    ((spiketime + window_starttime - signal.t_start) * signal.sampling_rate).simplified))
+                # adds the signal in selected interval relative to the spike
+                result_sta[:, i] += signal[startbin: startbin + window_bins, i]
+                # counting of the used spikes
+                used_spikes[i] += 1
+            else:
+                # counting of the unused spikes
+                unused_spikes[i] += 1
+
+        # normalization
+        result_sta[:, i] = result_sta[:, i] / used_spikes[i]
+
+        total_used_spikes += used_spikes[i]
+
+    if total_used_spikes == 0:
+        raise ValueError(
+            "No spike at all was either found or used for averaging")
```

Comment:
I personally don't think that this is the case where an error should be raised.
What's the argument against just returning zeros as STA in this case (as the function does in the case where one of given multiple spike trains happens to have no spike to be used for the averaging)?